internal/reconcile/step.go

```diff
@@ -92,3 +101,35 @@ func StepFailed(err error) Step {
 		Error:         err,
 	}
 }
+
+func StepWarning(err error) Step {
```

Comment:
lets add comments explaining the new steps purposes.

Reply:
of course